mfa_helper/login.sh

```diff
@@ -6,69 +6,114 @@
 #
 # Dependencies: ./jq
 # https://stedolan.github.io/jq/
-#
+
+set -e
+
+trap 'echo -e "\nAborted due to error" && exit 1' ERR
+trap 'echo -e "\nAborted by user" && exit 1' SIGINT
+
+readonly _CONFIG_FILE="$HOME/.aws/custom_config_options"
+
+main() {
+    parse_opts
+    detect_os
+    load_config
+
+    update_credentials
+
+    if [ "$DOCKER_FLAG" = true ]; then
+        docker_login
+    fi
+}
+
 print_usage() {
     echo "don't ever speak to me or my son again"
 }
 
-while getopts 'DP:' flag; do
-    case "${flag}" in
-        D) DOCKER_FLAG='true' ;;
-        P) AWS_PROFILE="${OPTARG}" ;;
-        *) print_usage
-           exit 1 ;;
+parse_opts() {
+    while getopts 'DP:' flag; do
+        case "${flag}" in
+            D) DOCKER_FLAG=true ;;
+            P) AWS_PROFILE="${OPTARG}" ;;
+            *) print_usage
+               exit 1 ;;
+        esac
+    done
+
+    if [ $DOCKER_FLAG = true ] && [[ -z $AWS_PROFILE ]]; then
```

Comment:
@dshoreman only error is here on my side. It complains about a unary operator expectation. Quotes around $DOCKER_FLAG fixes this. Will merge after that change.

Reply:
Heh, quoted one but not the other! Thanks, fixed